src/utils.jl

```diff
@@ -0,0 +1,37 @@
+
+function _bisection(f, left, right; max_iters=10000, tol=1e-10)
```

Comment:
The code in this function will yield many iterations or not converge because of strict inequality check, one should use:
`a ≈ b atol=tol` instead of equality checks for floating-point numbers

Reply:
Originally designed to hit floating point limit (since intervals are relatively small), but I agree that this level of precision is probably yields too many iterations. Will change to `abs(f(mid)) < tol` for early stop (similar to lines 32-33 for `max_iter` stop check).

Reply:
> Originally designed to hit floating point limit

yes but here you ask for strict equality right?

Reply:
Yes, so the algorithm stops when the boundary is indistinguishable from the midpoint to floating point precision. E.g., `1  == 1 + eps()/2` evaluates to `true`.